### ISSUE TITLE
Questions on how to print out training loss for each epoch

### ISSUE THREAD

**Issue by hohaxu** [Microsoft/LightGBM#1296]
Hello, Sorry this may be a very basic question. But I have spent a few hours on it and searched stackoverflow etc :(

Is there a way to print out training loss for each epoch? I tried setting 'is_training_metric' = True and seems not working..

**Comment by guolinke** [comment]
`is_training_metric=True` should work. 
Which package/version you are using with ? And did  you give the metric type ?

**Comment by hohaxu** [comment]
**version should be 2.1.0**

>>> import lightgbm
>>> print lightgbm.__version__
2.1.0

--------
**config as following**

params = {
    'task': 'train',
    'boosting_type': 'gbdt',
    'objective': 'binary',
    'metric': {'l2', 'auc', 'binary'},
    'is_training_metric': True,
    'metric_freq': 5,
    'num_leaves': 31,
    'learning_rate': 0.5,
    'feature_fraction': 0.9,
    'bagging_fraction': 0.8,
    'bagging_freq': 5,
    'verbose': 1,
    'device' : 'cpu',
    'gpu_platform_id' : 0,
    'gpu_device_id' : 0
}

---------
**Training log as following:**

Load data...
Start training...
[LightGBM] [Info] Number of positive: 475, number of negative: 2525
[LightGBM] [Info] Total Bins 70025
[LightGBM] [Info] Number of data: 3000, number of used features: 1781
[1]     valid_0's l2: 0.128616  valid_0's auc: 0.538306 valid_0's binary_logloss: 0.436842
Training until validation scores don't improve for 5 rounds.
[2]     valid_0's l2: 0.0965723 valid_0's auc: 0.543287 valid_0's binary_logloss: 0.350661
[3]     valid_0's l2: 0.0901258 valid_0's auc: 0.547033 valid_0's binary_logloss: 0.325112
[4]     valid_0's l2: 0.0888109 valid_0's auc: 0.538325 valid_0's binary_logloss: 0.3204
[5]     valid_0's l2: 0.0886666 valid_0's auc: 0.536894 valid_0's binary_logloss: 0.321005
[6]     valid_0's l2: 0.0880843 valid_0's auc: 0.545004 valid_0's binary_logloss: 0.323217
[7]     valid_0's l2: 0.0889384 valid_0's auc: 0.562401 valid_0's binary_logloss: 0.328373
[8]     valid_0's l2: 0.0901687 valid_0's auc: 0.545799 valid_0's binary_logloss: 0.340926
[9]     valid_0's l2: 0.0894441 valid_0's auc: 0.546881 valid_0's binary_logloss: 0.345127
Early stopping, best iteration is:
[4]     valid_0's l2: 0.0888109 valid_0's auc: 0.538325 valid_0's binary_logloss: 0.3204

**Comment by guolinke** [comment]
you should use 
```
gbm = lgb.train(
params,
lgb_train,
num_boost_round=20,
valid_sets=[lgb_train, lgb_eval],
early_stopping_rounds=5)
```

```is_training_metric=True``` only works in cli version.

**Comment by hohaxu** [comment]
Yes it works! Thanks a lot!
